mycore-ocfl/src/main/java/org/mycore/ocfl/metadata/migration/MCROCFLCombineComparePruner.java

```diff
@@ -0,0 +1,111 @@
+ * revision that is the result of merging the two given revisions.
+ */
+public abstract class MCROCFLCombineComparePruner implements MCROCFLRevisionPruner {
+
+    private static final Logger LOGGER = LogManager.getLogger();
+
+    @Override
+    public List<MCROCFLRevision> prune(List<MCROCFLRevision> revisions) throws IOException, JDOMException {
+        ArrayList<MCROCFLRevision> newRevisions = new ArrayList<>();
+        Iterator<MCROCFLRevision> iterator = revisions.listIterator();
+
+        if (!iterator.hasNext()) {
+            return newRevisions;
+        }
+
+        MCROCFLRevision last = iterator.next();
+        RevisionMergeDecider comparator = getMergeDecider();
+
+        while (iterator.hasNext()) {
+            MCROCFLRevision next = iterator.next();
+
+            if ((last.getType() == MCROCFLVersionType.CREATE || last.getType() == MCROCFLVersionType.UPDATE) &&
+                    (next.getType() == MCROCFLVersionType.CREATE || next.getType() == MCROCFLVersionType.UPDATE)) {
+                MCRContent currentContent = last.getContentSupplier().get();
+                Document currentDocument = currentContent.asXML();
+
+                MCRContent nextContent = next.getContentSupplier().get();
+                Document nextDocument = nextContent.asXML();
+
+                if (comparator.shouldMerge(last, currentDocument, next,nextDocument)) {
+                    LOGGER.info("Merging revisions {} and {}", last, next);
+                    last = buildMergedRevision(last, next, currentDocument, nextDocument);
+                    continue;
+                }
+            }
+
+            LOGGER.info("Keeping revision {}", last);
+            newRevisions.add(last);
+            last = next;
+        }
+
+        LOGGER.info("Keeping revision {}", last);
+        newRevisions.add(last);
+
+        return newRevisions;
+    }
+
+    /**
+     * Returns a {@link RevisionMergeDecider} that decides if two revisions should be merged.
+     * @return a {@link RevisionMergeDecider}
+     */
+    public abstract RevisionMergeDecider getMergeDecider();
+
+    /**
+     * Builds a new revision that is the result of merging the two given revisions.
+     * @param current the current revision
+     * @param next the next revision to merge
+     * @param currentDocument the current document
+     * @param nextDocument the document of the next revision
+     * @return a new revision that is the result of merging the two given revisions
+     */
+    public abstract MCROCFLRevision buildMergedRevision(MCROCFLRevision current, MCROCFLRevision next,
+        Document currentDocument, Document nextDocument);
+
+    /**
+     * A decider that decides if two revisions should be merged.
+     */
+    public interface RevisionMergeDecider {
+
+        boolean shouldMerge(MCROCFLRevision r1, Document o1, MCROCFLRevision r2, Document o2);
```

Comment:
use full names. r1 maybe understandable, but o1 should be document1. This should be fixed in other parts of the code too.